File: jdroid-android-core/src/main/java/com/jdroid/android/ActionItem.java
package com.jdroid.android;

import android.content.Intent;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import java.io.Serializable;

public interface ActionItem extends Serializable {

	String getName();

	Integer getIconResource();

	Integer getNameResource();

	Integer getDescriptionResource();

	Fragment createFragment(Object args);

	void startActivity(FragmentActivity fragmentActivity);

	Intent getIntent();

	Boolean matchesActivity(FragmentActivity fragmentActivity);
}

File: jdroid-android-core/src/main/java/com/jdroid/android/ActionItem.kt
package com.jdroid.android

import android.content.Intent
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import java.io.Serializable

interface ActionItem : Serializable {

    fun getId(): String

    fun getIconResource(): Int

    fun getNameResource(): Int

    fun getDescriptionResource(): Int?

    fun createFragment(args: Any): Fragment?

    fun startActivity(fragmentActivity: FragmentActivity)

    fun getIntent(): Intent

    fun matchesActivity(fragmentActivity: FragmentActivity): Boolean
}
